src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1838,7 +1838,9 @@ abstract class RefChecks extends Transform {
         )
         if (!isOk) {
           val msg = s"side-effecting nullary methods are discouraged: suggest defining as `def ${sym.name.decode}()` instead"
-          val namePos = sym.pos.focus.withEnd(sym.pos.point + sym.decodedName.length)
+          val namePos =
+            if (sym.pos.isRange) sym.pos
+            else sym.pos.toRange.withEnd(sym.pos.point + sym.decodedName.length)
```

Comment:
`sym.pos` is the `tree.namePos`. What about names from duplicated trees in for comprehensions? Pos of dup tree is focused, but does it inherit `NamePos`? (Lint uses start to correlate those symbols.)